util-core/src/main/scala/com/twitter/util/Future.scala

```diff
@@ -514,8 +518,8 @@ def join[%s](%s): Future[(%s)] = join(Seq(%s)) map { _ => (%s) }""".format(
     f: Seq[In] => Future[Seq[Out]]
   )(
     implicit timer: Timer
-  ): In => Future[Out] = {
-    new BatchExecutor[In, Out](sizeThreshold, timeThreshold, sizePercentile, f)
+  ): Batcher[In, Out] = {
```

Comment:
So how do you want to proceed here?  I don't have any insight into how often you can or want to do major releases.  I'm happy to add another endpoint if that helps this get through faster, but overall I think returning a more full-featured object is the way to go in the end.